src/Microsoft.Extensions.Localization/ResourceManagerStringLocalizerFactory.cs

```diff
@@ -65,12 +66,27 @@ public IStringLocalizer Create(Type resourceSource)
 
             var typeInfo = resourceSource.GetTypeInfo();
             var assembly = typeInfo.Assembly;
+            var assemblyName = new AssemblyName(assembly.FullName);
+
+            var resourcePath = _resourcesRelativePath;
+            var baseNamespace = _hostingEnvironment.ApplicationName;
+
+            //This isn't the main assembly
+            if (!string.Equals(assemblyName.Name, _hostingEnvironment.ApplicationName, StringComparison.Ordinal))
+            {
+                var resourceLocationAttribute = assembly.GetCustomAttribute<ResourceLocationAttribute>();
+                // If we don't have an attribute, just assume it's in 'Resources'
+                resourcePath = string.IsNullOrEmpty(resourceLocationAttribute?.ResourceFolder)
+                    ? "Resources"
```

Comment:
After thinking about this, I think the right thing to do here, rather than just assuming the folder is "Resources", is to do basically what we do for project resources and leave it null so we get TypeInfo.FullName.